src/test/java/com/squareup/javapoet/AnnotatedTypeNameTest.java

```diff
@@ -137,4 +140,79 @@ private void annotatedEquivalence(TypeName type) {
     String actual = type.toString();
     assertEquals(expected, actual);
   }
+
+  // https://github.com/square/javapoet/issues/614
+  @Ignore @Test public void annotatedArrayType() {
+    String expected =
+        "java.lang.Object @" + TypeUseAnnotation.class.getCanonicalName() + " []";
```

Comment:
Should this introduce `TUA` and `TYPE_USE_ANNOTATION` constants like the `NN` and `NEVER_NULL` above? That would probably make the tests a bit easier to read. WDYT?

Reply:
Good idea.